### PR TITLE
[expo-cli][config] EAS Build tweaks

### PR DESCRIPTION
**Pull request by dsokal**
I made two improvements which are split into two commits:
- 1st adds a feature that initializes a git repository for the project if it doesn't exist yet.
- 2nd improves reading the bundle identifier from the Xcode project. With my changes, it's possible to resolve the bundle id from the project initialized with `npx react-native init`. The id is resolved from the build configuration for the first native target available for the Xcode project.

### CODE REVIEW

File: packages/config/src/ios/BundleIdentifier.ts [hunk 1]
```diff
@@ -56,10 +52,25 @@ function getBundleIdentifierFromPbxproj(projectRoot: string): string | null {
...
+
+  for (const [, item] of getBuildConfigurationForId(project, nativeTarget.buildConfigurationList)) {
+    const bundleIdentifierRaw = item.buildSettings.PRODUCT_BUNDLE_IDENTIFIER;
+    if (bundleIdentifierRaw) {
+      const bundleIdentifier =
+        bundleIdentifierRaw[0] === '"' ? bundleIdentifierRaw.slice(1, -1) : bundleIdentifierRaw;
+      const bundleIdentifierParts = bundleIdentifier.split('.');
+      if (
+        bundleIdentifierParts[bundleIdentifierParts.length - 1] ===
+          '$(PRODUCT_NAME:rfc1034identifier)' &&
```
Comment (wkozyra95) [inline packages/config/src/ios/BundleIdentifier.ts:66]: maybe add some comment what it is

**Review by wkozyra95 — APPROVED** [review]

File: packages/expo-cli/src/commands/eas-build/build/utils/git.ts [hunk 2]
```diff
@@ -1,11 +1,59 @@
...
+    name: 'message',
+    message: 'Commit message:',
+    initial: 'Initial commit',
+    validate: input => input !== '',
+  });
+  await spawnAsync('git', ['add', '-A']);
+  await spawnAsync('git', ['commit', '-m', message]);
+}
+
+async function doesGitRepoExist(): Promise<boolean> {
```
Comment (wkozyra95) [inline packages/expo-cli/src/commands/eas-build/build/utils/git.ts:48]: You could move this to ` packages/expo-cli/src/git.ts`

File: packages/expo-cli/src/commands/eas-build/build/utils/git.ts [hunk 3]
```diff
@@ -1,12 +1,51 @@
 import spawnAsync from '@expo/spawn-async';
 import fs from 'fs-extra';
 import ora from 'ora';
+import which from 'which';
```
Comment (wkozyra95) [inline packages/expo-cli/src/commands/eas-build/build/utils/git.ts:4]: there is already `command-exists` package added to expo-cli deps that could be used here

https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/credentials/views/SetupAndroidKeystore.ts

**Review by wkozyra95 — APPROVED** [review]

**Comment by brentvatne** [comment]
great job on this on @dsokal :)

**Pull request merged**